libs/iplant-clojure-commons/src/clojure_commons/exception.clj

```diff
@@ -26,118 +22,189 @@
+
+(def failed-dependency-handler
+  (default-error-handler resp/failed-dependency ec/ERR_MISSING_DEPENDENCY))
+
+(def forbidden-handler
+  (default-error-handler resp/forbidden ec/ERR_FORBIDDEN {"WWW-Authenticate" "Custom"}))
+
+(def handle-request-validation-errors
+  (as-de-exception-handler ex/request-validation-handler ec/ERR_ILLEGAL_ARGUMENT))
+
+(def handle-response-validation-errors
+  (as-de-exception-handler ex/response-validation-handler ec/ERR_SCHEMA_VALIDATION))
+
+(def illegal-argument-handler
+  (default-error-handler resp/bad-request ec/ERR_ILLEGAL_ARGUMENT))
+
+(def invalid-cfg-handler
+  (default-error-handler resp/internal-server-error ec/ERR_CONFIG_INVALID))
+
+(def invalid-json-handler
+  (default-error-handler resp/bad-request ec/ERR_INVALID_JSON))
+
+(def item-exists-handler
+  (default-error-handler resp/bad-request ec/ERR_EXISTS))
+
+(def missing-query-params-handler
+  (default-error-handler resp/bad-request ec/ERR_MISSING_QUERY_PARAMETER))
+
+(def missing-request-field-handler
+  (default-error-handler resp/bad-request ec/ERR_BAD_OR_MISSING_FIELD))
+
+(def not-authorized-handler
+  (default-error-handler resp/unauthorized ec/ERR_NOT_AUTHORIZED))
+
+(def not-found-handler
+  (default-error-handler resp/not-found ec/ERR_NOT_FOUND))
+
+(def not-owner-handler
+  (default-error-handler resp/forbidden ec/ERR_NOT_OWNER))
+
+(def not-writeable-handler
+  (default-error-handler resp/bad-request ec/ERR_NOT_WRITEABLE))
+
+(def not-a-user-handler
+  (default-error-handler resp/not-found ec/ERR_NOT_A_USER))
+
+(def request-failed-handler
+  (default-error-handler resp/internal-server-error ec/ERR_REQUEST_FAILED))
+
+(def unavailable-handler
+  (default-error-handler resp/gateway-timeout ec/ERR_UNAVAILABLE))
 
 (defn unchecked-handler
-  [error error-type _]
+  [error error-type request]
   (let [error-obj (:object (get-throw-context error))]
     (cond
-     (ec/error? error-obj)
-     (embed-error-info error
-                       error-obj
-                       (resp/internal-server-error (cheshire/encode error-obj)))
-
-     (clj-http-error? error-obj)
-     (embed-error-info error
-                       {:error_code ec/ERR_REQUEST_FAILED}
-                       error-obj)
+      (ec/error? error-obj)
+      (let [exception error-obj
+            response (resp/internal-server-error (cheshire/encode error-obj))]
+        (smw/log-response-with-exception error
+                                         request
+                                         response
+                                         exception)
+        response)
+
+      (clj-http-error? error-obj)
+      (let [exception {:error_code ec/ERR_REQUEST_FAILED}
+            response error-obj]
+        (smw/log-response-with-exception error
+                                         request
+                                         response
+                                         exception)
+        response)
```

Comment:
Rereview always being what it is as far as noticing new things... might it make sense to have `log-response` (and thus `log-response-with-exception`) just return the response so you don't need to have this last line in all these functions? Or is that too close to being magic and thus avoided?